src/DatePicker/Calendar.js

```diff
@@ -81,14 +83,19 @@ class Calendar extends Component {
         selectedDate: date,
       });
     }
+    if (nextProps.displayMode !== this.props.displayMode) {
```

Comment:
We are using a hybrid approach too with the `<Snackbar />` and the `<Drawer />` components. They are uncontrolled by default and can become controlled if users want to.
I think that we need the same behavior here. Still, we could improve the implementation by:
- making our component stateless by default.
- having an extra layer to handle the uncontrolled / controlled switch and the needed state.